packages/mui-lab/src/LoadingButton/LoadingButton.js

```diff
@@ -101,17 +101,27 @@ const LoadingButtonLoadingIndicator = styled('div', {
   position: 'absolute',
   visibility: 'visible',
   display: 'flex',
-  ...(ownerState.loadingPosition === 'start' && {
-    left: 14,
-  }),
+  ...(ownerState.loadingPosition === 'start' &&
+    (ownerState.variant === 'outlined' || ownerState.variant === 'contained') && {
+      left: 14,
+    }),
+  ...(ownerState.loadingPosition === 'start' &&
+    (ownerState.variant === 'text' || !ownerState.variant) && {
```

Comment:
Could we please default the `variant` to `text`? It is more explicit. Now we depend that the `Button` will have that variant as default implicitly.